File: petshop.cpp
#include <iostream>
#include <list>
#include <string>

using namespace std;

class petshop
{
private:
  int id;
  string nama;
  string kategori;
  int harga;
  list<petshop> produk;

public:
  petshop() : id(0), nama(""), kategori(""), harga(0) {}

  petshop(int id, string nama, string kategori, int harga) : id(id), nama(nama), kategori(kategori), harga(harga) {}

  void setId(int id) { this->id = id; }
  void setNama(string nama) { this->nama = nama; }
  void setKategori(string kategori) { this->kategori = kategori; }
  void setHarga(int harga) { this->harga = harga; }

  int getId() { return id; }
  string getNama() { return nama; }
  string getKategori() { return kategori; }
  int getHarga() { return harga; }

  void tampilkan()
  {
    cout << "ID: " << id << " | Nama: " << nama << " | Kategori: " << kategori << " | Harga: " << harga << endl;
  }

  void tambahData()
  {
    int harga;
    string nama, kategori;
    cin.ignore();
    cout << "Masukkan Nama: ";
    getline(cin, nama);
    cout << "Masukkan Kategori: ";
    getline(cin, kategori);
    cout << "Masukkan Harga: ";
    cin >> harga;

    int id = produk.size() + 1; // ID berdasarkan jumlah elemen

    // Buat objek dan atur nilai
    petshop item;
    item.setId(id);
    item.setNama(nama);
    item.setKategori(kategori);
    item.setHarga(harga);

    produk.push_back(item); // Tambahkan ke list produk milik objek
    cout << "Data berhasil ditambahkan dengan ID " << id << "!\n";
  }

  void tampilkanData()
  {
    if (produk.empty())
    {
      cout << "Tidak ada data." << endl;
      return;
    }

    for (auto &item : produk)
    {
      item.tampilkan();
    }
  }

  void ubahData()
  {
    int id;
    cout << "Masukkan ID yang ingin diubah: ";
    cin >> id;

    for (auto &item : produk)
    {
      if (item.getId() == id)
      {
        string nama, kategori;
        int harga;
        cin.ignore();
        cout << "Masukkan Nama baru: ";
        getline(cin, nama);
        cout << "Masukkan Kategori baru: ";
        getline(cin, kategori);
        cout << "Masukkan Harga baru: ";
        cin >> harga;
        item.setNama(nama);
        item.setKategori(kategori);
        item.setHarga(harga);
        cout << "Data berhasil diubah!\n";
        return;
      }
    }

    cout << "Data tidak ditemukan!\n";
  }

  void hapusData()
  {
    int id;
    cout << "Masukkan ID yang ingin dihapus: ";
    cin >> id;

    for (auto it = produk.begin(); it != produk.end(); ++it)
    {
      if (it->getId() == id)
      {
        produk.erase(it);
        cout << "Data berhasil dihapus!\n";
        return;
      }
    }

    cout << "Data tidak ditemukan!\n";
  }

  void cariData()
  {
    string nama;
    cin.ignore();
    cout << "Masukkan Nama yang dicari: ";
    getline(cin, nama);

    for (auto &item : produk)
    {
      if (item.getNama() == nama)
      {
        item.tampilkan();
        return;
      }
    }

    cout << "Data tidak ditemukan!\n";
  }
};

File: PYTHON/PetShop.py
class PetShop:
    def __init__(self, id=0, nama="", kategori="", harga=0):
        self.id = id
        self.nama = nama
        self.kategori = kategori
        self.harga = harga
        self.produk = []

    def set_id(self, id):
        self.id = id

    def set_nama(self, nama):
        self.nama = nama

    def set_kategori(self, kategori):
        self.kategori = kategori

    def set_harga(self, harga):
        self.harga = harga

    def get_id(self):
        return self.id

    def get_nama(self):
        return self.nama

    def get_kategori(self):
        return self.kategori

    def get_harga(self):
        return self.harga

    def tampilkan(self):
        print(f"ID: {self.id} | Nama: {self.nama} | Kategori: {self.kategori} | Harga: {self.harga}")

    def tambah_data(self):
        nama = input("Masukkan Nama: ")
        kategori = input("Masukkan Kategori: ")
        harga = int(input("Masukkan Harga: "))
        
        id_baru = len(self.produk) + 1
        item = PetShop(id_baru, nama, kategori, harga)
        self.produk.append(item)
        print(f"Data berhasil ditambahkan dengan ID {id_baru}!")

    def tampilkan_data(self):
        if not self.produk:
            print("Tidak ada data.")
            return
        
        for item in self.produk:
            item.tampilkan()

    def ubah_data(self):
        id_ubah = int(input("Masukkan ID yang ingin diubah: "))
        for item in self.produk:
            if item.get_id() == id_ubah:
                nama = input("Masukkan Nama baru: ")
                kategori = input("Masukkan Kategori baru: ")
                harga = int(input("Masukkan Harga baru: "))
                item.set_nama(nama)
                item.set_kategori(kategori)
                item.set_harga(harga)
                print("Data berhasil diubah!")
                return
        print("Data tidak ditemukan!")

    def hapus_data(self):
        id_hapus = int(input("Masukkan ID yang ingin dihapus: "))
        for i, item in enumerate(self.produk):
            if item.get_id() == id_hapus:
                del self.produk[i]
                print("Data berhasil dihapus!")
                return
        print("Data tidak ditemukan!")

    def cari_data(self):
        nama_cari = input("Masukkan Nama yang dicari: ")
        for item in self.produk:
            if item.get_nama() == nama_cari:
                item.tampilkan()
                return
        print("Data tidak ditemukan!")
